lighthouse-core/gather/gatherers/dobetterweb/all-event-listeners.js

```diff
@@ -118,18 +118,16 @@ class EventListeners extends Gatherer {
     })).then(nestedListeners => [].concat(...nestedListeners));
   }
 
-  beforePass(options) {
-    this.driver = options.driver;
-    this._parsedScripts = new Map();
-    return this.listenForScriptParsedEvents();
-  }
-
   /**
    * @param {!Object} options
    * @return {!Promise<!Array<!Object>>}
    */
   afterPass(options) {
-    return this.unlistenForScriptParsedEvents()
+    this.driver = options.driver;
+    this._parsedScripts = new Map();
+    return Promise.resolve()
+      .then(() => this.listenForScriptParsedEvents())
+      .then(() => this.unlistenForScriptParsedEvents())
       .then(_ => options.driver.querySelectorAll('body, body /deep/ *')) // drill into shadow trees
```

Comment:
if you want to fix the 1st half of https://github.com/GoogleChrome/lighthouse/issues/2163
we're awfully close to it right now.

i'm also fine with it coming in separately.